### PR TITLE
[CHORE] Upgrade version to v0.9.32

### LINKED ISSUE #402: [Chore] Update kvdb-rocksdb dependency
Blocked by: #705 
- Next release of Substrate (Polkadot `v0.9.31`)
- Might get fixed with #705 

Update the crates.io dependency of https://crates.io/crates/kvdb-rocksdb when an update is released.

We wish to include the latest updates to the crates `kvdb-rocksdb` package. Parity plans on pushing a release to this package at some point in the future; when they do, we should pin our `kvdb-rocksdb` dependency from current:`0.15.2` to this version.

## Notes
- `kvdb-rocksdb` does not use `rocksdb_open_column_families_with_ttl` so is not effected by https://github.com/advisories/GHSA-xpp3-xrff-w6rh
- Is effected by https://github.com/advisories/GHSA-9qxh-258v-666c but the release of 0.16.0 removes that package.

## Acceptance Criteria
- [ ] Update `kvdb-rocksdb` to the future release
- [ ] Update ci/deny.toml

### LINKED ISSUE #519: Switch to use Rayon for check_delegations
Blocked By: Substrate v0.9.31/32

[`check_delegations`](https://github.com/LibertyDSNP/frequency/blob/be72f52d8fb2d84451ba5bf55224f6c07ca3b2c5/pallets/msa/src/rpc/src/lib.rs#L68-L88) uses an `iter`, but we want to use a Rayon `par_iter`, however that currently panics. It is corrected in a [Merged PR](https://github.com/paritytech/substrate/pull/12502), but it is not in a release yet.

https://substrate.stackexchange.com/questions/5329/panic-when-using-rayon-with-runtime-api-in-a-custom-rpc/5373#5373

### LINKED ISSUE #705: Upgrade to v0.9.32
Blocked by #704 and Milestone https://github.com/LibertyDSNP/frequency/milestone/30

## Known ToDos
- [x] https://github.com/paritytech/polkadot/releases/tag/v0.9.32
- [x] Bump Cargo.toml versions
- [x] Likely bump runtime version

### LINKED ISSUE #865: [Vulnerabilities]Dependabot alerts: rocksdb vulnerable to out-of-bounds read
See:
[rocksdb vulnerable to out-of-bounds read](https://github.com/LibertyDSNP/frequency/security/dependabot/14)

### LINKED ISSUE #866: [Vulnerabilities]Dependabot alerts: owning_ref vulnerable to multiple soundness issues
See:
[owning_ref vulnerable to multiple soundness issues ](https://github.com/LibertyDSNP/frequency/security/dependabot/13)

### PR DESCRIPTION
**Pull request by saraswatpuneet**
# Goal
The goal of this PR is 

Closes #705 
Closes #402 
Closes #519 
Closes #866
Closes #865

Notes:

[PR for Substrate upgrade to 0.9.32](https://github.com/substrate-developer-hub/substrate-parachain-template/pull/138)
[Release Notes .31](https://github.com/paritytech/polkadot/releases/tag/v0.9.31)
[Release Notes .32](https://github.com/paritytech/polkadot/releases/tag/v0.9.32)

# Changes from 0.9.30 via 0.9.31 to 0.9.32

- Bump dependency version to 0.9.32
- Weight V2 : introduction of max proof size for weights, for now I am using what cumulus uses and suggested on substrate stack exchange ...[more](https://substrate.stackexchange.com/questions/5557/construct-runtime-integrity-test-failing).
- Runtime builds now run a runtime integrity checks 
- Major Democracy Pallet upgrades and changes

# Checklist
- [x] Design doc(s) updated
- [x] Weights updated
- [x] Run benchmarks

### CODE REVIEW

**Comment by saraswatpuneet** [comment]
Branched out from #744, Ideally review after it merges

File: runtime/common/src/constants.rs [hunk 1]
```diff
@@ -96,8 +96,9 @@ pub const AVERAGE_ON_INITIALIZE_RATIO: Perbill = Perbill::from_percent(5);
 pub const NORMAL_DISPATCH_RATIO: Perbill = Perbill::from_percent(75);
 
 /// We allow for 0.5 of a second of compute with a 12 second average block time.
-pub const MAXIMUM_BLOCK_WEIGHT: Weight = WEIGHT_PER_SECOND.saturating_div(2);
-
+pub const MAXIMUM_BLOCK_WEIGHT: Weight = WEIGHT_PER_SECOND
+	.saturating_div(2)
+	.set_proof_size(cumulus_primitives_core::relay_chain::v2::MAX_POV_SIZE as u64);
```
Comment (saraswatpuneet) [inline runtime/common/src/constants.rs:101]: Something to learn about , this starts transition into weights v2
Currently other parachains using the one set by cumulus (as we are cumulus based parachain)

File: .github/workflows/merge-pr.yml [hunk 2]
```diff
@@ -94,6 +94,10 @@ jobs:
     steps:
       - name: Check Out Repo
         uses: actions/checkout@v3
+      - name: Install Required Packages
+        run: |
+          sudo apt-get update
+          sudo apt install -y protobuf-compiler
       - name: Install Rust Toolchain
         uses: actions-rs/toolchain@v1
```
Comment (saraswatpuneet) [inline .github/workflows/merge-pr.yml:102]: Substrate needs protoc library this pretty much became a requirement starting .3

File: integration-tests/package.json [hunk 3]
```diff
@@ -13,8 +13,8 @@
     "author": "",
     "license": "Apache-2.0",
     "dependencies": {
-        "@frequency-chain/api-augment": "v0.9.29",
-        "@polkadot/api": "9.6.2",
+        "@frequency-chain/api-augment": "^v0.9.29",
```
Comment (saraswatpuneet) [inline integration-tests/package.json:16]: Keeping dependency for integration tests pinned to first stable release .9.29

File: pallets/msa/src/rpc/src/lib.rs [hunk 4]
```diff
@@ -89,17 +90,16 @@ where
...
-		let at = BlockId::hash(self.client.info().best_hash);
-
-		Ok(delegator_msa_ids
-			.iter() // TODO: Change back to par_iter() which has borrow panic GitHub Issue: #519
-			.map(|&delegator_msa_id| {
+		let results = delegator_msa_ids
+			.par_iter()
+			.map(|delegator_msa_id| {
+				let api = self.client.runtime_api();
+				let at = BlockId::hash(self.client.info().best_hash);
```
Comment (saraswatpuneet) [inline pallets/msa/src/rpc/src/lib.rs:97]: API is not thread safe so we can use rayon for this iteration

File: pallets/msa/src/rpc/src/lib.rs [hunk 5]
```diff
@@ -89,17 +90,16 @@ where
...
-		let api = self.client.runtime_api();
 		let at = BlockId::hash(self.client.info().best_hash);
-
-		Ok(delegator_msa_ids
-			.iter() // TODO: Change back to par_iter() which has borrow panic GitHub Issue: #519
-			.map(|&delegator_msa_id| {
+		let results = delegator_msa_ids
+			.par_iter()
+			.map(|delegator_msa_id| {
+				let api = self.client.runtime_api();
```
Comment (saraswatpuneet) [inline pallets/msa/src/rpc/src/lib.rs:97]: Using rayon to run this iteration now that runtime api is threadsafe

File: node/service/src/service.rs [hunk 6]
```diff
@@ -191,10 +191,8 @@ async fn build_relay_chain_interface(
 	hwbench: Option<sc_sysinfo::HwBench>,
 ) -> RelayChainResult<(Arc<(dyn RelayChainInterface + 'static)>, Option<CollatorPair>)> {
 	match collator_options.relay_chain_rpc_url {
-		Some(relay_chain_url) => {
-			let client = create_client_and_start_worker(relay_chain_url, task_manager).await?;
-			Ok((Arc::new(RelayChainRpcInterface::new(client)) as Arc<_>, None))
-		},
+		Some(relay_chain_url) =>
+			build_minimal_relay_chain_node(polkadot_config, task_manager, relay_chain_url).await,
```
Comment (saraswatpuneet) [inline node/service/src/service.rs:195]: Change introduce in how clients are created now, pretty excited to see how relay can be ported out of collator

File: integration-tests/package-lock.json [hunk 7]
```diff
@@ -1,40 +1,70 @@
 {
     "name": "frequency-integration-tests",
     "version": "1.0.0",
-    "lockfileVersion": 1,
+    "lockfileVersion": 3,
```
Comment (saraswatpuneet) [inline integration-tests/package-lock.json:4]: Had to update my npm to be on lockfile version 3 same as some other devs

File: rust-toolchain.toml [hunk 8]
```diff
@@ -1,5 +1,5 @@
 [toolchain]
-channel = "nightly-2022-09-22"
+channel = "nightly-2022-10-09"
```
[the remark this answers is not in the document]
Comment (saraswatpuneet) [reply rust-toolchain.toml:2]: this overrides whatever we set in workflows

File: pallets/msa/src/lib.rs [hunk 9]
```diff
@@ -108,7 +108,7 @@ pub mod weights;
 
 #[frame_support::pallet]
 pub mod pallet {
-	use frame_support::log::error;
+	use frame_support::log::error as log_err;
```
[the remark this answers is not in the document]
Comment (saraswatpuneet) [reply pallets/msa/src/lib.rs:111]: this started happening with .32
https://github.com/LibertyDSNP/frequency/actions/runs/3625144913/jobs/6112861027

File: runtime/frequency/src/lib.rs [hunk 10]
```diff
@@ -388,7 +388,7 @@ parameter_types! {
 	pub MaximumSchedulerWeight: Weight = Perbill::from_percent(10) * RuntimeBlockWeights::get().max_block;
 	// The maximum number of scheduled calls in the queue for a single block. Not strictly enforced, but used for weight estimation.
 	// Retry a scheduled item every 25 blocks (5 minute) until the preimage exists.
-	// Will be removed in v0.9.30
+	// Will be removed in v0.9.32
 	pub const NoPreimagePostponement: Option<u32> = Some(5 * MINUTES);
```
[the remark this answers is not in the document]
Comment (saraswatpuneet) [reply runtime/frequency/src/lib.rs]: Just removed it

File: rust-toolchain.toml (hunk 8 above)
[the remark this answers is not in the document]
Comment (saraswatpuneet) [reply rust-toolchain.toml:2]: so in workflows we are on stable toolchain always and this will decide which toolchain we prefer

**Review by wilwade — APPROVED** [review]
🎉

**Review by enddynayn — APPROVED** [review]
💯 These look so painful. I salute you.

**Pull request merged**